### DISCUSSION TITLE
Rename "internal" to "cause" on `RpcError`

### DISCUSSION THREAD

**Discussion by kasparkallas** [wevm/wagmi#871] [Ideas]
Seems like the `.internal` is semantically the same as `.cause` which is becoming a standardized way to chain errors.

Some references:
* `error.cause` browser support (part of ES2022): https://caniuse.com/mdn-javascript_builtins_error_cause
* `error.cause` added to Node 16.9: https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V16.md#16.9.0
* `error.cause` added to Node documentation: https://github.com/nodejs/node/pull/43830
* `error.cause` added to TypeScript: https://github.com/microsoft/TypeScript/issues/45167

**Comment by tmm** [comment]
Good idea! Added in https://github.com/wagmi-dev/wagmi/pull/1412
